adapters/pubnative/pubnative.go

```diff
@@ -0,0 +1,170 @@
+package pubnative
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type PubnativeAdapter struct {
+	URI string
+}
+
+func (a *PubnativeAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	impCount := len(request.Imp)
+	requestData := make([]*adapters.RequestData, 0, impCount)
+	errs := []error{}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	err := checkRequest(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	for _, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		var pubnativeExt openrtb_ext.ExtImpPubnative
+		if err := json.Unmarshal(bidderExt.Bidder, &pubnativeExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		err := convertImpression(&imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		reqJSON, err := json.Marshal(request)
```

Comment:
Here you are looping over the imps, modifying one `Imp` in each loop, and then generating a request for each `Imp` that includes all the `Imp`s, with the current and prior `Imp`s modified, and the remaining `Imp`s untouched. You probably want to make a shallow copy of `request`, and assign its `Imp[]` as a one element slice containing the current modified `Imp` that you are looping over.

Reply:
You're right, thanks for spotting this 👍